src/scripts/quests/Quest.ts

```diff
@@ -215,6 +233,48 @@ abstract class Quest {
         this.autoCompleter?.dispose();
     }
 
+    withDescription(description: string): Quest {
+        this.customDescription = description;
+        return this;
+    }
+
+    withOnLoad(onLoad: () => void): Quest {
+        this._onLoad = onLoad;
+        return this;
+    }
+
+    withCustomReward(customReward: () => void): Quest {
+        this.customReward = typeof customReward === 'function' ? customReward : undefined;
+        return this;
+    }
+
+    withOptionalArgs(optionalArgs: QuestOptionalArgument): Quest {
+        this.optionalArgs = optionalArgs;
+        return this;
+    }
+
+    withInitialValue(initialValue: number): Quest {
+        this.initialValue = initialValue;
+        return this;
+    }
+
+    public getClearedMessage() {
+        return this.optionalArgs.clearedMessage;
+    }
+
+    public getNpcDisplayName() {
+        return this.optionalArgs.npcDisplayName;
+    }
+
+    public getNpcImageName() {
```

Comment:
This can be removed because npc images are determined through `getNpcImage` below.